Wizardry/Core/BattleSys/Source/BattleOrder.c

```diff
@@ -418,6 +419,9 @@ int GetBattleUnitHitCount(struct BattleUnit * actor)
                                ? &gBattleTarget
                                : &gBattleActor;
 
+    if (GetCombatArtInForce(&actor->unit) == CID_Detonate)
```

Comment:
no need this

Reply:
Could you explain why? I need some way to ensure that the user can only attack once with Detonate.

Reply:
https://github.com/MokhaLeee/fe8u-cskillsys-kernel/blob/main/Wizardry/Core/BattleSys/Source/BattleOrder.c#L32

Reply:
Combat arts are inherently designed not to double attack by default, just as what ThreeHouses did

Reply:
As I said, the likes of DoubleLion etc override this. If you have a better solution, I'll remove my code for this and let you do it. 

![image](https://github.com/user-attachments/assets/cc4f240e-fcb9-4f09-a9e6-b63f91f4134c)

![image](https://github.com/user-attachments/assets/ab871791-ee73-42f7-9f72-e7561ecb1e03)

Reply:
.It is just a feature with no conflict between "double-lion attack twice" and "combat-art cannot double". The former get more hits inside each round, but the latter just limit the round amount to ONE.

Reply:
It conflicts with the skill itself. If the user 'detonates' their equipped weapon, they shouldn't be able to attack twice.

Reply:
Maybe it'd be best to create a global flag for these situations? Something like, `solo_attack = true` for when a unit shouldn't gain any additional attacks from their speed or skills.

Reply:
I think the real question need to think about is why the heros weapon can handle the hit count correctly but here failed.

Reply:
Although it has been fixed by https://github.com/MokhaLeee/fe8u-cskillsys-kernel/commit/6cb08bc9118728ce2ee565e58025cd14e57b233b, it still hold bugs if you directly set the weapon usage to 0xFF, since function `BattleForecastHitCountUpdate()` cannot handle on it. For now I also have no good idea on solve it. I think you can pull the skill/combat-art info and comment out the skill index first. Later we can modify on basic functionality to adapt it.

Reply:
Made the changes. 

On a slighly different note, what's your plan to deal with skills like Shove or GoddessDance, as they're not combat related, so won't make sense as arts.

Reply:
No plan for now, maybe add a seperate menu command later